src/zenml/integrations/great_expectations/data_validators/ge_data_validator.py

```diff
@@ -323,126 +306,17 @@ def root_directory(self) -> str:
-            ValueError: if an `expectation_suite_name` value is not supplied and
-                a name for the expectation suite cannot be generated from the
-                current step name and pipeline name.
-        """
-        context = self.data_context
-
-        if comparison_dataset is not None:
-            logger.warning(
-                "A comparison dataset is not required by Great Expectations "
-                "to do data profiling. Silently ignoring the supplied dataset "
-            )
-
-        if not expectation_suite_name:
-            try:
-                step_context = get_step_context()
-                pipeline_name = step_context.pipeline.name
-                step_name = step_context.step_run.name
-                expectation_suite_name = f"{pipeline_name}_{step_name}"
-            except RuntimeError:
-                raise ValueError(
-                    "A expectation suite name is required when not running in "
-                    "the context of a pipeline step."
-                )
-
-        suite_exists = False
-        if context.expectations_store.has_key(  # noqa
-            ExpectationSuiteIdentifier(expectation_suite_name)
-        ):
-            suite_exists = True
-            suite = context.get_expectation_suite(expectation_suite_name)
-            if not overwrite_existing_suite:
-                logger.info(
-                    f"Expectation Suite `{expectation_suite_name}` "
-                    f"already exists and `overwrite_existing_suite` is not set "
-                    f"in the step configuration. Skipping re-running the "
-                    f"profiler."
-                )
-                return suite
-
-        batch_request = create_batch_request(context, dataset, data_asset_name)
-
-        try:
-            if suite_exists:
-                validator = context.get_validator(
-                    batch_request=batch_request,
-                    expectation_suite_name=expectation_suite_name,
-                )
-            else:
-                validator = context.get_validator(
-                    batch_request=batch_request,
-                    create_expectation_suite_with_name=expectation_suite_name,
-                )
-
-            profiler = UserConfigurableProfiler(
-                profile_dataset=validator, **profiler_kwargs
-            )
-
-            suite = profiler.build_suite()
-            context.save_expectation_suite(
-                expectation_suite=suite,
-                expectation_suite_name=expectation_suite_name,
-            )
-
-            context.build_data_docs()
-        finally:
-            context.delete_datasource(batch_request.datasource_name)
-
-        return suite
-
     def data_validation(
         self,
         dataset: pd.DataFrame,
         comparison_dataset: Optional[Any] = None,
         check_list: Optional[Sequence[str]] = None,
+        expectations_list: Optional[Sequence[GreatExpectationExpectationConfig]] = None,
```

Comment:
As far as I can see, you can provide either an `expectation_suite_name` or an `expectations_list` to choose how we will validate the dataset.

If I understand correctly, if an `expectation_suite_name` is given we use the `context.suite` to fetch the correct `suite` but if an `expectations_list` is given we create a new `suite` with the `f"{run_name}_{step_name}"` pattern.

If that's the case, both docstrings are misleading as they say `A value must be provided` even though you just have to provide one of them.

Reply:
yes, you understand it exactly right. i'll fix the docstrings.